lib/async_hooks.js

```diff
@@ -0,0 +1,307 @@
+'use strict';
+
+const async_wrap = process.binding('async_wrap');
+// A Uint32Array wrapping the int64_t current_async_id_. Used as the JS
+// equivalent of node::Environment::current_async_id().
+const current_id_array = async_wrap.current_id_array;
+// A Uint32Array wrapping the int64_t async_uid_. Used as the JS equivalent
+// of node::Environment::new_async_uid().
+const new_id_array = async_wrap.new_id_array;
+// A Uint32Array wrapping node::Environment::AsyncHooks::fields_. An array that
+// communicates between JS and C++ how many of a given type of callback are
+// available to be called.
+const async_hook_fields = async_wrap.async_hook_fields;
+// Array of all AsyncHooks that will be iterated whenever an async event fires.
+const active_hooks_array = [];
+const isBE = require('os').endianness() === 'BE';
```

Comment:
Ugh. Can we please consider making this a constant in `os.constants` rather than a method?